src/libraries/System.Private.CoreLib/src/System/IO/Strategies/BufferedFileStreamStrategy.cs

```diff
@@ -604,6 +604,7 @@ private void WriteByteSlow(byte value)
             }
             else
             {
+                Debug.Assert(_writePos == _bufferSize - 1);
```

Comment:
https://helix.dot.net/api/2019-06-17/jobs/36aff9ac-be62-40c1-8ced-40871cec2272/workitems/System.IO.FileSystem.Tests/console

I'll send a separate PR to address that

```suggestion
                Debug.Assert(_writePos <= _bufferSize);
```